NullPointerException in IterablePushActionReceiver

When we click on the notification, If the app is closed, we get crash. 

- Both 3.0.0 & 3.0.1 versions of SDK

- Android 7.0 & 8.0 & 8.1

- When the app is on background, It works fine. We can navigate to the app. However, If the app is not on background(closed) the app crashes when notification clicked. 

```
java.lang.RuntimeException: Unable to start receiver com.iterable.iterableapi.IterablePushActionReceiver: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.SharedPreferences android.content.Context.getSharedPreferences(java.lang.String, int)' on a null object reference
        at android.app.ActivityThread.handleReceiver(ActivityThread.java:3194)
        at android.app.ActivityThread.-wrap17(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1672)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.SharedPreferences android.content.Context.getSharedPreferences(java.lang.String, int)' on a null object reference
        at com.iterable.iterableapi.IterableApi.getPreferences(IterableApi.java:1193)
        at com.iterable.iterableapi.IterableApi.setAttributionInfo(IterableApi.java:115)
        at com.iterable.iterableapi.IterableApi.setNotificationData(IterableApi.java:172)
        at com.iterable.iterableapi.IterablePushActionReceiver.handlePushAction(IterablePushActionReceiver.java:84)
        at com.iterable.iterableapi.IterablePushActionReceiver.onReceive(IterablePushActionReceiver.java:39)
        at android.app.ActivityThread.handleReceiver(ActivityThread.java:3187)
```
Any help would be appreciated.

--- a/iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java
+++ b/iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java
@@ -111,6 +111,11 @@ public class IterableApi {
      * @param attributionInfo Attribution information object
      */
     void setAttributionInfo(IterableAttributionInfo attributionInfo) {
+        if (_applicationContext == null) {
+            IterableLogger.e(TAG, "setAttributionInfo: Iterable SDK is not initialized with a context.");
+            return;
+        }
+
         IterableUtil.saveExpirableJsonObject(
                 getPreferences(),
                 IterableConstants.SHARED_PREFS_ATTRIBUTION_INFO_KEY,

--- a/iterableapi/src/test/java/com/iterable/iterableapi/IterablePushActionReceiverTest.java
+++ b/iterableapi/src/test/java/com/iterable/iterableapi/IterablePushActionReceiverTest.java
@@ -62,6 +62,19 @@ public class IterablePushActionReceiverTest extends BaseTest {
     }
 
     @Test
+    public void testPushOpenWithNonInitializedSDK() throws Exception {
+        stubAnyRequestReturningStatusCode(200, "{}");
+        IterableApi.sharedInstance = new IterableApi();
+        IterablePushActionReceiver iterablePushActionReceiver = new IterablePushActionReceiver();
+        Intent intent = new Intent(IterableConstants.ACTION_PUSH_ACTION);
+        intent.putExtra(IterableConstants.ITERABLE_DATA_ACTION_IDENTIFIER, "silentButton");
+        intent.putExtra(IterableConstants.ITERABLE_DATA_KEY, IterableTestUtils.getResourceString("push_payload_silent_action.json"));
+
+        // This must not crash
+        iterablePushActionReceiver.onReceive(RuntimeEnvironment.application, intent);
+    }
+
+    @Test
     public void testTrackPushOpenWithCustomAction() throws Exception {
         final JSONObject responseData = new JSONObject("{\"key\":\"value\"}");
         stubAnyRequestReturningStatusCode(200, responseData);

